lib: Improved warnings for 'no (enable) password'

ppmathis opened the pull request:
When the user executes one of the commands 'no password' or 'no enable
password', a warning message gets shown to inform the user of the
security implications.

While the current implementation works, a warning message gets printed
once for each daemon, which can lead to seeing the same message many
times. This does not affect functionality, but looks like an error to
the user as it can be seen within issue #1432.

This commit only prints the warning message inside lib when vtysh
dispatch is not being used. Additionally, the warning message was copied
into the vtysh command handlers, so that they get printed exactly once.

Fixes #1432 

Signed-off-by: Pascal Mathis <mail@pascalmathis.com>

qlyoung commented:
@ppmathis How about making the error message a `#define` in `command.h`, like `#define NO_PASSWD_CMD_WARNING "Please be aware..."` and using that everywhere? Alternatively, just change the strings to be on one line (it's okay if it's over 80 characters for user-visible strings) and I'll merge it.

ppmathis commented:
@qlyoung Sure, that's actually a great idea. I've moved the string into `command.h` and removed the linebreaks within the user-visible string. Should be all good now according to clang-format.
